CRM/Contribute/BAO/ContributionRecur.php

```diff
@@ -94,6 +96,43 @@ public static function add(&$params) {
     return $recurring;
   }
 
+  /**
+   * Event fired after modifying a contribution.
+   * @param \Civi\Core\Event\PostEvent $event
+   */
+  public static function self_hook_civicrm_post(\Civi\Core\Event\PostEvent $event) {
+    if ($event->action === 'edit') {
+      if (is_numeric($event->object->amount)) {
+        $templateContribution = CRM_Contribute_BAO_ContributionRecur::getTemplateContribution($event->object->id);
+        if (empty($templateContribution['id']) || isset(\Civi::$statics[__CLASS__]['update'][$templateContribution['id']])) {
+          // Precaution against a loop - although in theory the lookups provide this
+          // it still seems worth an extra check.
+          return;
+        }
+        $lines = LineItem::get(FALSE)
+          ->addWhere('contribution_id', '=', $templateContribution['id'])
+          ->addWhere('contribution_id.is_template', '=', TRUE)
+          ->addSelect('contribution_id.total_amount')
+          ->execute();
+        if (count($lines) === 1) {
+          $contributionAmount = $lines->first()['contribution_id.total_amount'];
+          if (Money::of($contributionAmount, 'USD')->compareTo($event->object->amount)) {
```

Comment:
I'm not so concerned with that scenario as much as seeing a red flag whenever currencies get hard-coded. Here's an example: https://lab.civicrm.org/dev/core/-/issues/2930 describing how if a price set gets applied to contributions of different currencies, all hell breaks loose.

Reply:
I *think* in this case 'USD is being used because the formatting matches how it's stored in object->amount. We definitely need a comment here explaining that if that is the case.

Reply:
right the use of currency is just to compare two numbers & the library requires them to have a currency  - anyone would do

Reply:
on looking I've cast them BOTH to USD & added a comment